src/attr/_make.py

```diff
@@ -1412,24 +1474,33 @@ def _make_eq(cls, attrs):
         "    if other.__class__ is not self.__class__:",
         "        return NotImplemented",
     ]
+
     # We can't just do a big self.x = other.x and... clause due to
     # irregularities like nan == nan is false but (nan,) == (nan,) is true.
+    global_vars = {}
     if attrs:
         lines.append("    return  (")
         others = ["    ) == ("]
         for a in attrs:
-            lines.append("        self.%s," % (a.name,))
-            others.append("        other.%s," % (a.name,))
+            if a.eq_key:
+                cmp_name = "_%s_key" % (a.name,)
+                # Add the cmp class to the global namespace of the evaluated
+                # function, since local does not work in all python versions.
```

Comment:
Could you evaluate please? Adding stuff to the global namespace sounds risky.

If this is a Python 2 thing, I'd rather make the feature Python 3-only.

Reply:
We can do that but there is a precedent: we do use the global dictionary in ``_make_attr_tuple_class`` and ``_make_init``.

Reply:
I would strongly prefer to make this Python 3-only then. pip just dropped Python 2, we don't need to bend backward anymore when it comes to new features.

Unless the code becomes even more complicated? I guess I'll leave this to you, but I'm -0 on global namespace shenanigans.